### PR TITLE
Adding missing rust examples

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

File: api-code-examples/rust/docs-join.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    print(f"Started Iroh node: {node_id}")
+
+    # you'll need to get a ticket from somewhere, see the `doc share` command documentation
+    # for details. This ticket will fail to join, but is a valid ticket.
+    TICKET = "docaaa7qg6afc6zupqzfxmu5uuueaoei5zlye7a4ahhrfhvzjfrfewozgybl5kkl6u6fqcnjxvdkoihq3nbsqczxeulfsqvatb2qh3bwheoyahacitior2ha4z2f4xxk43fgewtcltemvzhaltjojxwqltomv2ho33snmxc6biajjeteswek4ambkabzpcfoajganyabbz2zplaaaaaaaaaagrjyvlqcjqdoaaioowl2ygi2likyov62rofk4asma3qacdtvs6whqsdbizopsefrrkx"
+
+    doc = await node.docs().join(TICKET)
+    print(f"Joined doc: {doc.id()}")
+
+asyncio.run(main())
```
Comment (matheus23) [inline api-code-examples/rust/docs-join.rs:18]: This file seems like it's pyhton, but stored as `docs-join.rs`, is that intentional?

File: api-code-examples/rust/docs-keys.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import iroh
+import asyncio
+
+
+async def main():
```
Comment (matheus23) [inline api-code-examples/rust/docs-keys.rs:5]: This one, too

Comment (matheus23) [reply api-code-examples/rust/docs-keys.rs:5]: (and more)

Comment (dignifiedquire) [reply api-code-examples/rust/docs-keys.rs:5]: see the draft state? working my way through these to convert

**Review by b5 — APPROVED** [review]
Zomg this is soooooo useful. Tyvm diggle

**Pull request merged**